packages/storage-plus/src/de.rs

```diff
@@ -0,0 +1,157 @@
+use crate::keys::IntKey;
+use cosmwasm_std::{Addr, StdError, StdResult};
+use std::array::TryFromSliceError;
+use std::convert::TryInto;
+
+pub trait Deserializable {
+    type Output: Sized;
+
+    fn from_slice(value: &[u8]) -> StdResult<Self::Output>;
+    fn from_vec(value: &Vec<u8>) -> StdResult<Self::Output>;
```

Comment:
`fn from_vec(value: &Vec<u8>)` is a very odd signature.
if you made it `fn from_vec(value: Vec<u8>)` I would understand this better (to avoid clone when you can grant ownership).

If this is not the purpose, then I think we need better documentation for the difference

Reply:
This is wip at the moment. My idea here was to pass a Vec to try to do a zero-copy / no-op impl in the case the Output is a Vec.

Clippy says this is more expensive than passing a reference to a slice, so, seems `from_slice` is the way to go. Will revert this.

The only way I can think of to try to improve on this, would be to use a macro instead of a function to do the conversion, and (somehow) do nothing in the case of Vec output.